### ISSUE TITLE
Image Descriptor class implementation

### ISSUE THREAD

**Issue by penny4860** [penny4860/object-detector#4]
Image Descriptor is needed to detect object in image.
Following descriptors will be implemented in concrete class
- Color Histogram
- HOG
- LBP

and they will have common interface.

**Comment by penny4860** [comment]
HOG descriptor is implemented and merged into master at 62f94ba9f5133b6b3b6c8af3e025f5317081553b. Other descriptor need to be implemented in the near future.

**Comment by penny4860** [comment]
Let's implement class architecture as follows

```
class Detector
  def __init__(descriptor = HOG(...), classifier = SVM(...)):
    self._descriptor = descriptor
    self._classifier = classifier

  def extract_feature():
    self._descriptor.describe() 
```
